lib/miq_automation_engine/engine/miq_ae_service/miq_ae_service_rbac.rb

```diff
@@ -0,0 +1,34 @@
+module MiqAeMethodService
+  module MiqAeServiceRbac
+    extend ActiveSupport::Concern
+
+    module ClassMethods
+      def find_ar_object_by_id(id)
+        Rbac.search(:class => model, :targets => [id],
+                    :results_format => :objects).try(:first).try(:first)
```

Comment:
@kbrock 
This code doesn't lend itself to passing in the :user object, even though we have it in the workspace. We have use cases where customer automate method scripts and our own internal code directly instantiate the ServiceModel objects. We set the User.current_user in the workspace at the beginning of the Automation Request based on the passed in user,group,tenant. We would have to create a global variable to store the user object and use it in all the places that try  to fetch/instantiate the service models. The User.current_user is the global variable in our case.